### ISSUE TITLE
Higher performance Fluent Bit Plugin available?

### ISSUE THREAD

**Issue by pontiphex** [aws/amazon-kinesis-streams-for-fluent-bit#235]
Hello!

I noticed on https://github.com/aws/amazon-kinesis-firehose-for-fluent-bit that there's a higher performance plugin written in C available (https://docs.fluentbit.io/manual/pipeline/outputs/firehose). Is it also true that the parallel output plugin for Data Streams (https://docs.fluentbit.io/manual/pipeline/outputs/kinesis) is higher performance than the Go plugin?

Only asking because I saw the notice on that repo but not this one and am curious if that will affect how we use the AWS for fluent bit image.

Thanks!

**Comment by ashraf133** [comment]
Hello
I haven't mesured performance but the GO plugin offers additionals features like : compression, aggregation ....

**Comment by PettitWesley** [comment]
Yes, the C/core plugin is higher performance and should have much lower resource usage, especially lower memory usage: https://docs.fluentbit.io/manual/pipeline/outputs/kinesis

But as noted, this repo (the go plugin) has some features which are not yet in the high performance version. So you have to make a decision based on which is more important for your use case.

**Comment by seasuresh** [comment]
@PettitWesley Is there a plan to incorporate the "Match," "Compression," and "Retry Limit" features from the existing plugin into the new "C" plugin? We are interested in using the new plugin, but are unable to do so because these features are currently absent.

**Comment by PettitWesley** [comment]
@seasuresh `Match` and `Retry_Limit` are basic Fluent Bit config options that work for all outputs because they are managed by the core engine. Compression support for `kinesis_streams` C plugin is a feature request.

**Comment by jas3kera** [comment]
Hi @PettitWesley,

I'm interested in these 2 unsupported features (compression and aggregation) for _kinesis_streams_ C plugin. I wonder if there is a roadmap and/or thinking about supporting it? Do you know?

Thank you in advance.
